site-scrapers/PriceChopper.js

```diff
@@ -0,0 +1,75 @@
+const sites = require("../data/sites.json");
+const https = require("https");
+const crypto = require('crypto');
+
+const siteName = 'Price Chopper';
+const site = sites[siteName];
+
+module.exports = async function GetAvailableAppointments(browser) {
+    console.log(`${siteName} starting.`);
+    const webData = await ScrapeWebsiteData(browser);
+    console.log(`${siteName} done.`);
+    return site.locations.map((loc) => {
+        const locHash = md5HashString(loc.street + loc.city);
+        const responseLocation = webData[locHash];
+        let hasAvailability = false;
+        let availability = {};
+        if(responseLocation && responseLocation.visibleTimeSlots) {
+            hasAvailability = responseLocation.visibleTimeSlots.length ? true : false;
+            for(let [i, timeSlot] of responseLocation.visibleTimeSlots.entries()) {
+                let date = timeSlot.time.split('T')[0];
+                if(!availability[date]){
+                    availability[date] = {
+                        hasAvailability: true,
+                        numberAvailableAppointments: 0,
+                        signUpLink: site.signUpLink
```

Comment:
(FYI: I know I should document this) we only put a signUpLink under a date's availability if there's a distinct sign-up link for each date (like for the MA sites). Since the link is the same here for all dates, we keep the signUpLink key on the top-level only (one for each location, not one for each date)

Reply:
Makes sense.